# Исследование надежности заемщиков


## Описание проекта

Заказчик — кредитный отдел банка.\
Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок.\
Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы,\
которая оценивает способность потенциального заёмщика вернуть кредит банку.

### Данные

Входные данные от банка — статистика о платёжеспособности клиентов.

**Описание данных**
- children — количество детей в семье
- days_employed — общий трудовой стаж в днях
- dob_years — возраст клиента в годах
- education — уроaвень образования клиента
- education_id — идентификатор уровня образования
- family_status — семейное положение
- family_status_id — идентификатор семейного положения
- gender — пол клиента
- income_type — тип занятости
- debt — имел ли задолженность по возврату кредитов
- total_income — ежемесячный доход
- purpose — цель получения кредита

### Задача

Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок.

Потребуется:
- изучить общую информацию о полученных данных, проанализировать какие данные количественные/качественные
- выявить и обработать пропущенные значения для каждого столбца,
- выявить и обработать аномальные значения,
- для удобства проведения анализа:
    - заменить некоторые типы данных на необходимые типы данных
    - категоризировать данные
- выявить и обработать явные и неявные дубликаты

Подтвердить или опровергнуть следующие гипотезы:
- Есть ли зависимость между количеством детей и возвратом кредита в срок?
- Есть ли зависимость между семейным положением и возвратом кредита в срок?
- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?
- Как разные цели кредита влияют на его возврат в срок?

### Инструменты библиотеки и техники
**Python, Pandas, предобработка данных, лемматизация**

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

Для выявления зависимостей воспользуемся показателями из описательной статистики для количественного описания данных:

- общее кол.во строк ('count'),
- сумма значений строк ('sum'),
- среднее ('mean'),

в разрезе интересуемого сегмента данных.

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Сгруппируем данные по столбцу 'children' и найдем показатели каждого полученного сегмента.

In [22]:
data_grouped = data.groupby('children').agg({'debt':['count','sum', 'mean']})
dict_to_rename = dict(zip(data_grouped.columns.levels[1], ['Общее кол.-во строк', 'Сумма значений','Среднее']))
data_grouped = data_grouped.rename(columns=dict_to_rename, level=1)
data_grouped

debt                         
         Общее кол.-во строк Сумма значений   Среднее
children                                             
0                      14091           1063  0.075438
1                       4808            444  0.092346
2                       2052            194  0.094542
3                        330             27  0.081818
4                         41              4  0.097561
5                          9              0  0.000000

**Вывод:**

Гипотеза подтвердилась:
> Имеется зависимость между количеством детей и возвратом кредита в срок:
>
> с увеличением количества детей мы видим увеличение количества просроченных задолженностей.

У бездетных просрочка оплаты ниже, чем у людей с детьми.

Люди с 3-мя детьми чаще платят в срок, чем люди с одним ребенком, двумя или четырьмя детьми.

Не выявлено ни одной задолженности для семей с 5-ю детьми, соответственно, доля задолженности равняется нулю.
С учетом этого, рекомендуется исследование на более широкой выборке.


#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Сгруппируем данные по столбцу 'family_status' и найдем показатели каждого полученного сегмента.

In [23]:
data_grouped = data.groupby('family_status').agg({'debt':['count','sum', 'min', 'max', 'median', 'mean']}).round(2)
dict_to_rename = dict(zip(data_grouped.columns.levels[1], ['Общее кол.-во строк', 'Сумма значений', 'Минимум', 'Максимум', 'Медиана', 'Среднее']))
data_grouped = data_grouped.rename(columns=dict_to_rename, level=1)
data_grouped

debt                                  \
                      Общее кол.-во строк Сумма значений Минимум Максимум   
family_status                                                               
Не женат / не замужем                2796            273       0        1   
в разводе                            1189             84       0        1   
вдовец / вдова                        951             63       0        1   
гражданский брак                     4134            385       0        1   
женат / замужем                     12261            927       0        1   

                                       
                      Медиана Среднее  
family_status                          
Не женат / не замужем       0    0.10  
в разводе                   0    0.07  
вдовец / вдова              0    0.07  
гражданский брак            0    0.09  
женат / замужем             0    0.08

**Вывод:**

> Имеется зависимость между семейным положением и возвратом кредита

Те кто развелись или овдовели чаще платят в срок, чем люди в браке.

Люди не в браке и не бывавшие в браке имеют больший процент невозвратов в срок.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Сгруппируем данные по столбцу 'total_income_category' и найдем показатели каждого полученного сегмента.

In [24]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

data_grouped = data.groupby('total_income_category').agg({'debt':['count','sum', 'mean']})
dict_to_rename = dict(zip(data_grouped.columns.levels[1], ['Общее кол.-во строк', 'Сумма значений', 'Среднее']))
data_grouped = data_grouped.rename(columns=dict_to_rename, level=1)
data_grouped

debt                         
                      Общее кол.-во строк Сумма значений   Среднее
total_income_category                                             
A                                      25              2  0.080000
B                                    5014            354  0.070602
C                                   15921           1353  0.084982
D                                     349             21  0.060172
E                                      22              2  0.090909

Оценим репрезентативность выборки по количеству участников каждого сегмента: A, B, C, D, E.

Всего участников: 25 + 5041 + 15921 + 349 + 22 = 21358
На один сегмент количество участников должно стремиться к среднему: 21358 / 5 = 4271.6

В этой связи, не берем в расчет категории A, D, E так как они представлены менее 10% от репрезентативной выборки:
(25 / 4271.6) или (350 / 4271.6) или (22/ 4271.6) < 10%.

По другому, интерпретировать этот результат можно как - среди участников выделяются только два значимых сегмента:
- до 200000 (это участники из категорий C, D, E)
- 200001 и более (A, B)

Таким образом, вывод будем делать на основе на категории B и C.

**Вывод:** 

> Имеется зависимость между уровнем дохода и возвратом кредита в срок:\
> наибольший риск стать должниками у категории C, а наименьший у категории B.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Сгруппируем данные по столбцу 'purpose_category' и найдем показатели каждого полученного сегмента.

In [25]:
data_grouped = data.groupby('purpose_category').agg({'debt':['count','sum', 'mean']})
dict_to_rename = dict(zip(data_grouped.columns.levels[1], ['Общее кол.-во строк', 'Сумма значений', 'Среднее']))
data_grouped = data_grouped.rename(columns=dict_to_rename, level=1)
data_grouped

debt                         
                         Общее кол.-во строк Сумма значений   Среднее
purpose_category                                                     
операции с автомобилем                  4279            400  0.093480
операции с недвижимостью               10751            780  0.072551
получение образования                   3988            369  0.092528
проведение свадьбы                      2313            183  0.079118

**Вывод:** 

> Цель кредита влияет на его возврат в срок

Чаще задолженность возникает по кредитам на "операции с автомобилем" и "получение образования".

Меньше задолженность возникает по кредитам "проведение свадьбы" и "операции с недвижимостью"



#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке.

Причинами ошибок могут быть:
- неверный ввод данных,
- сокрытие информации,
- повреждение файла данных при сохранении,
- технические проблемы.

На практике пропуски в данных — повод поговорить с разработчиками. Лучше остановить исследование, узнать причину пропусков и устранить её. И тогда исследовать уже данные без пропущенных значений.

Как мы убедились выше (п.2.1 Задание 4.), пропуски наблюдаются в столбцах 'days_employed' и 'total_income' в 2174 строках.

Убедимся, что для одних и тех же строк, одновременно отсутствуют данные по обоим столбцам:

In [26]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')
    
data[(data['days_employed'].isna() == True) & (data['total_income'].isna() == True)].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2174 entries, 12 to 21510
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          2174 non-null   int64  
 1   days_employed     0 non-null      float64
 2   dob_years         2174 non-null   int64  
 3   education         2174 non-null   object 
 4   education_id      2174 non-null   int64  
 5   family_status     2174 non-null   object 
 6   family_status_id  2174 non-null   int64  
 7   gender            2174 non-null   object 
 8   income_type       2174 non-null   object 
 9   debt              2174 non-null   int64  
 10  total_income      0 non-null      float64
 11  purpose           2174 non-null   object 
dtypes: float64(2), int64(5), object(5)
memory usage: 220.8+ KB


**Ответ:**

Предположение подтвердилось: если в строке отсутствуют данные в столбце 'days_employed', то одновременно отсутствуют данные и по 'total_income'.

Таким образом, это системный сбой, нужно сообщить разработчикам выявленную закономерность.


#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

**Ответ:**

Среднее арифметическое — это сумма всех значений, поделённая на количество значений.
Среднее арифметическое отражает ситуацию некорректно, так как не учитывает разброс данных.

Медиана — это такое число в выборке, что ровно половина элементов больше него, а другая половина — меньше.
На нее не влияют случайные выбросы и шум.
К тому же, если данные имеют нормальное распределение и их количество велико, то медиана будет равняться среднему значению.

В этой связи, медианное значение лучше всего отражает выборку в целом, в случае, если есть незаполненные данные.

### Шаг 4: общий вывод.

Входные данные от банка — статистика о платёжеспособности клиентов.

Все выявленные аномальные значения столбца 'days_employed' (содержали отрицательное количество дней трудового стажа) были обработаны и заменены на положительные с помощью метода abs().

Пропуски значений были найдены в двух столбцах 'days_employed' и 'total_income' в 2174 строках.
Причем, если в строке отсутствуют данные в столбце 'days_employed', то одновременно отсутствуют данные и по 'total_income'.
Либо это человеческий фактор, либо техническая ошибка, - в любом случае стоит более внимательно проанализировать эту закономерность,
и принять меры, чтобы избежать пропущенных значений при сборе данных в будущем.

Для заполнения пропущенных значений были использованы медианные значения, так как медиана менее чувствительна к выбросам, чем среднее значение.
Так как на сумму дохода 'total_income' сильнее всего влияет тип занятости,
поэтому в этом столбце пропуски были заполнены медианным значением по каждому типу из столбца 'income_type'.

После привидения строковых значений столбца 'education' к нижнему регистру были выявлены и удалены 71 строка неявных дубликатов.
Дубликаты строковых значений столбца 'education' - это следствие заполнения произвольным строковым значением.
В этой связи, при сборе данных в будущем рекомендуется ограничивать выбор пользователя из списка возможных значений.

Для возможности делать статистические выводы потребовалось дополнительно категоризировать данные.

Среди участников выделены два значимых сегмента по по уровню доходов:
- до 200000 
- 200001 и более

И четыре по целям кредита:
- ‘операции с автомобилем’,
- ‘операции с недвижимостью’,
- ‘проведение свадьбы’,
- ‘получение образования’.

Для оценки способности потенциального заёмщика вернуть кредит банку проведены исследования гипотез:

1. Зависимость между наличием детей и возвратом кредита в срок.

Гипотеза подтвердилась, с увеличением количества детей мы видим увеличение количества просроченных задолженностей.

- У бездетных просрочка оплаты ниже, чем у людей с детьми.
- Люди с 3-мя детьми чаще платят в срок, чем люди с одним ребенком, двумя или четырьмя детьми.
- Не выявлено ни одной задолженности для семей с 5-ю детьми, соответственно, доля задолженности равняется нулю.
С учетом этого, рекомендуется исследование на более широкой выборке.

2. Зависимость между семейным положением и возвратом кредита в срок

Имеется зависимость:
- Те кто развелись или овдовели чаще платят в срок, чем люди в браке.
- Люди не в браке и не бывавшие в браке имеют больший процент невозвратов в срок.

3. Зависимость между уровнем дохода и возвратом кредита в срок.

Имеется зависимость, - наибольший риск стать должниками у участников с уровнем доходов до 200000, а наименьший у участников с уровнем доходов 200001 и более.

4. Как разные цели кредита влияют на его возврат в срок
- Чаще задолженность возникает по кредитам на "операции с автомобилем" и "получение образования".
- Меньше задолженность возникает по кредитам "проведение свадьбы" и "операции с недвижимостью".